src/components/NavBar/ShoppingCart.tsx

```diff
@@ -1,16 +1,30 @@
 import React from "react";
 import { IconButton, Circle, Stack } from "@chakra-ui/react";
 import { HiOutlineShoppingBag } from "react-icons/hi";
+import { switchCartDisplay } from "@/store/reducer/cartControlSlice";
+import { useDispatch } from "react-redux";
 
 interface Props {
   quantity: number;
   loginState: boolean;
 }
 
 const ShoppingCart = ({ quantity, loginState }: Props) => {
+  const dispatch = useDispatch();
+  const handleCartPageOpen = () => {
+    if (loginState) {
```

Comment:
What if the user is not logged in?

Reply:
> What if the user is not logged in?

Currently nothing will happen when you click the shopping cart button if log-in state is false. I think we can create a new card for alerting some message or simply pop up the log in window.